src/feature/nodelist/networkstatus.c

```diff
@@ -1625,7 +1628,12 @@ routerstatus_has_changed(const routerstatus_t *a, const routerstatus_t *b)
          a->is_valid != b->is_valid ||
          a->is_possible_guard != b->is_possible_guard ||
          a->is_bad_exit != b->is_bad_exit ||
-         a->is_hs_dir != b->is_hs_dir;
+         a->is_hs_dir != b->is_hs_dir ||
```

Comment:
We've added a new flag since this ticket was last modified:
* is_staledesc

We also missed these fields last time, or the implementation has changed:
* has_bandwidth